x-pack/plugins/security_solution/scripts/endpoint/common/fleet_services.ts

```diff
@@ -209,9 +215,68 @@ export const waitForHostToEnroll = async (
     );
   }
 
-  log.debug(`Host [${hostname}] has been enrolled with fleet`);
+  log.info(`Host [${hostname}] has been enrolled with fleet`);
   log.verbose(found);
 
+  if (!awaitMetadata) {
+    return found;
+  }
+  log.info('Querying indices');
+
+  log.debug(
+    JSON.stringify(
+      await esClient?.search({
+        index: [
+          '.fleet-agents',
+          'metrics-endpoint.metadata_current_default',
+          '.metrics-endpoint.metadata_united_default',
+        ],
+      }),
+      null,
+      2
+    )
+  );
+  log.info('Successfully queried indices.');
+
+  log.info(`Awaiting host to show up in Metadata`);
+
+  const metadataLookupStartedTime = new Date();
+  const hasTimedOutMetadataLookup = (): boolean => {
+    const elapsedTime = Date.now() - metadataLookupStartedTime.getTime();
+    return elapsedTime > timeoutMs;
+  };
+  let metadataFound: HostInfo | undefined;
+
+  while (!metadataFound && !hasTimedOutMetadataLookup()) {
+    metadataFound = await retryOnError(
+      async () =>
+        fetchEndpointMetadataList(kbnClient).then((response) => {
```

Comment:
maybe not a huge deal since it seems unlikely to be used that way but if we use this function when >100 endpoints, there's a chance this won't work depending on how pagination works out. would be more efficient and safer to pass `hostname` and `host_status` filters directly into the [kuery param](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/endpoint/metadata/list_metadata_route.ts#L18) similar to [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/scripts/endpoint/agent_emulator/screens/policy_responder.ts#L121).